### PR TITLE
PAL-1086-data-reader-reactive-streams

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: NOTICES.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ List of third-party dependencies grouped by their license type
 * spring-boot-starter ([org.springframework.boot:spring-boot-starter:2.3.1.RELEASE](https://spring.io/projects/spring-boot))
 * spring-boot-starter-test ([org.springframework.boot:spring-boot-starter-test:2.3.1.RELEASE](https://spring.io/projects/spring-boot))
 * GCHQ Synthetic Data Generator ([uk.gov.gchq.data-gen:synthetic-data-generator:0.0.4](https://github.com/gchq/synthetic-data-generator))
+* GCHQ Palisade - Avro Serialiser ([uk.gov.gchq.palisade:avro-serialiser:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-readers/tree/develop/s3-data-reader))
```
Comment (developer01189998819991197253) [inline NOTICES.md:9]: This will need rebuilding to get updated to 0.5.1

File: deployment-k8s/pom.xml [hunk 2]
```diff
@@ -186,6 +186,13 @@
                         </goals>
                         <configuration>
                             <artifactItems>
+                                <artifactItem>
+                                    <groupId>${project.groupId}</groupId>
+                                    <artifactId>avro-serialiser</artifactId>
+                                    <version>0.5.0-${readers.revision}</version>
```
Comment (developer01189998819991197253) [inline deployment-k8s/pom.xml:192]: dependency versions need updating to use 5.0.1

File: example-runner/pom.xml [hunk 3]
```diff
@@ -69,6 +69,12 @@
             <artifactId>common</artifactId>
             <version>0.5.1-${common.revision}</version>
         </dependency>
+        <dependency>
+            <!-- User, Resource, Context types -->
+            <groupId>${project.parent.groupId}</groupId>
+            <artifactId>avro-serialiser</artifactId>
+            <version>0.5.0-${readers.revision}</version>
```
Comment (developer01189998819991197253) [inline example-runner/pom.xml:76]: ditto here

File: performance/pom.xml [hunk 4]
```diff
@@ -64,6 +64,12 @@
             <artifactId>common</artifactId>
             <version>0.5.1-${common.revision}</version>
         </dependency>
+        <dependency>
+            <!-- User, Resource, Context types -->
+            <groupId>${project.parent.groupId}</groupId>
+            <artifactId>avro-serialiser</artifactId>
+            <version>0.5.0-${readers.revision}</version>
```
Comment (developer01189998819991197253) [inline performance/pom.xml:71]: ditto

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

File: deployment-jvm/values.yaml [hunk 5]
```diff
@@ -17,9 +17,11 @@ replicaCount: 1
 image:
   name: deployment-jvm
   base: jdk
-  tag: SNAPSHOT-35667b3
+  tag: SNAPSHOT-693786a
   pullPolicy: IfNotPresent
-  codeRelease: 0.5.0-SNAPSHOT
+  versionNumber: 0.5.0
```
Comment (pd104923) [inline deployment-jvm/values.yaml:22]:
```suggestion
  versionNumber: 0.5.1
```

File: deployment-k8s/charts/deployment-k8s/values.yaml [hunk 6]
```diff
@@ -17,9 +17,9 @@ replicaCount: 1
 image:
   name: deployment-k8s
   base: jdk
-  tag: SNAPSHOT-0b838de
+  tag: SNAPSHOT-693786a
   pullPolicy: IfNotPresent
-  gitHash: 0b838de
+  gitHash: 693786a
   versionNumber: 0.5.0
```
Comment (pd104923) [inline deployment-k8s/charts/deployment-k8s/values.yaml:23]:
```suggestion
  versionNumber: 0.5.1
```

File: deployment-k8s/charts/example-runner/values.yaml [hunk 7]
```diff
@@ -21,10 +21,10 @@ service:
 image:
   name: example-runner
   base: jdk
-  tag: SNAPSHOT-0b838de
+  tag: SNAPSHOT-693786a
   pullPolicy: IfNotPresent
   codeRelease: 0.5.1-SNAPSHOT
-  gitHash: 0b838de
+  gitHash: 693786a
   versionNumber: 0.5.0
```
Comment (pd104923) [inline deployment-k8s/charts/example-runner/values.yaml:28]:
```suggestion
  versionNumber: 0.5.1
```

File: deployment-k8s/charts/performance/values.yaml [hunk 8]
```diff
@@ -21,10 +21,10 @@ service:
 image:
   name: performance
   base: jdk
-  tag: SNAPSHOT-0b838de
+  tag: SNAPSHOT-693786a
   pullPolicy: IfNotPresent
   codeRelease: 0.5.1-SNAPSHOT
-  gitHash: 0b838de
+  gitHash: 693786a
   versionNumber: 0.5.0
```
Comment (pd104923) [inline deployment-k8s/charts/performance/values.yaml:28]:
```suggestion
  versionNumber: 0.5.1
```

**Review by pd104923 — CHANGES_REQUESTED** [review]
A few little things

**Review by developer01189998819991197253 — APPROVED** [review]

**Pull request merged**